### ISSUE TITLE
Automated Rasberry Pi images

### ISSUE THREAD

**Issue by spuder** [scottgchin/delta5_race_timer#54]
It would be nice to have a pre-built raspberry pi image. 

pi-gen is the tool used to generate the pre built images on raspberrypi.org 

https://github.com/RPi-Distro/pi-gen

**Comment by scottgchin** [comment]
Sorry if I'm a bit slow, but help me understand what this does?
it creates an .img for easy installation?

**Comment by spuder** [comment]
Correct, so instead of people needing to install the software manually, they just download a .img to a sd card, plug it in and everything is up and running. 

Some example projects that ship pre-built images for raspberry-pi. 

[octopi](https://octopi.octoprint.org/)
[astrobox](https://astroprint.zendesk.com/hc/en-us/articles/202638365-Raspberry-Pi-Flashing-the-AstroBox-image-onto-an-SD-card)

I'm interested in helping with this.

**Comment by spuder** [comment]
The home assistant (hasbian) raspberry pi image is the best example. Steps to implement:

1. Fork the pi-gen image (I'll do this in my own fork for now, eventually it may be best to move it under a delta 5 organization)
2. Remove folders [stage 3,4,5] replace with own scripts to install and configure app
3. setup travis-ci to build a new image based on a schedule/event (weekly?, on merge to master?, whenever you feel like it)
4. Update documentation showing where to download images ([hasbian images](https://github.com/home-assistant/pi-gen/tree/v1.31) are 300-400mb which should be hostable on github)
5. Profit??

**Comment by scottgchin** [comment]
Cool.  I'll take a look at it when I get a chance.  I already posted the latest image of my SD card on the facebook page to make it easier for people to install.

**Comment by spuder** [comment]
I've nearly got the automated images working. They are on my delta5 branch in my fork. The bulk of the script is in step 3

https://github.com/spuder/pi-gen/tree/delta5

It is copied from the home-assistant project 

https://github.com/home-assistant/pi-gen

**Comment by spuder** [comment]
I've successfully got automated images building on my repo

https://github.com/spuder/pi-gen

    ./build-docker.sh

Depending on your internet speed, you will then have an image in about 30 minutes. 

The image is idential to a stock raspbian with the following changes. 

- source is cloned to /opt/delta5
- dedicated user to run service
- locale = US not GB (still buggy)
- nginx reverse proxy to forward port 80 to 5000
- i2c kernel modules

It will build a 456MB file. Use a program like etcher to drop it on an sd card, plug it in and wait 5 minutes and delta 5 will be running. 

## Next steps

- Transfer git repo to delta 5 project ownership
- setup travis-ci to build images every time delta 5 has a new release
- document how to use

**Comment by spuder** [comment]
I've got an image up on github. The biggest advantage to it is that it has UK keyboard and locale settings. 

https://github.com/spuder/pi-gen/releases

**Comment by scottgchin** [comment]
Sorry about the delay, Just now starting to go through this.  This looks pretty cool!  So how is the image creation triggered?  Is it each time there is a release?

**Comment by spuder** [comment]
That part still hasn’t been automated yet. Currently you need to install
Docker and then run the build-docker.sh script.

I’ve manually uploaded two images to github so far

Depending on how frequently delta 5 is released will determine how to
automate the raspberry pi image with Travis ci.

**Comment by h4rr21** [comment]
Hi spuder, I understand that part of the idea it's to make easy to deploy de code to the raspi. 

But what about "https://resin.io" shouldn't be easy to deploy if delta5 runs in a docker container with resin.io rather than uploading a complete new image each time it releases a new one?

BTW I'm interested in helping to adopt CI tools in this project...

**Comment by spuder** [comment]
There may be a place for resin.io, however that is more intended for IOT projects. Delta 5 is very often run without internet access. Also delta5 doesn't run in a docker container currently, it only uses docker to build the image that goes on the SD card. 

As far as I'm concerned, the raspberry pi images are working, you can get them here:

https://github.com/spuder/pi-gen/releases

Since delta 5 isn't releasing new versions very often, there isn't a need net for a more automated update process (yet).

**Comment by h4rr21** [comment]
What's the login user and password? For this images?

**Comment by BoLtWD** [comment]
> There may be a place for resin.io, however that is more intended for IOT projects. Delta 5 is very often run without internet access. Also delta5 doesn't run in a docker container currently, it only uses docker to build the image that goes on the SD card.
> 
> As far as I'm concerned, the raspberry pi images are working, you can get them here:
> 
> https://github.com/spuder/pi-gen/releases
> 
> Since delta 5 isn't releasing new versions very often, there isn't a need net for a more automated update process (yet).

This is fantastic work to help someone get going as setting up the Pi can be daunting.
